lib/code_corps/accounts/accounts.ex

```diff
@@ -29,8 +34,107 @@ defmodule CodeCorps.Accounts do
   def create_from_github_changeset(struct, %{} = params) do
     struct
     |> Changeset.change(params |> Adapters.User.from_github_user())
-    |> Changeset.put_change(:context, "github")
+    |> Changeset.put_change(:sign_up_context, "github")
     |> Changeset.unique_constraint(:email)
     |> Changeset.validate_inclusion(:type, ["bot", "user"])
   end
+
+  @doc ~S"""
+  Ensures an email is set without overwriting an existing email.
+  """
+  @spec ensure_email(Changeset.t, map) :: Changeset.t
+  def ensure_email(%Changeset{} = changeset, %{"email" => new_email} = _params) do
```

Comment:
The reason for the comment above is because we do the same here

The only place we call `ensure_email/2` from is `update_from_github_changeset/2`. The only reason it's public is so we can test it directly. 

`Accounts.Changesets` would, again, be a good candidate to do this instead.